core/src/main/java/com/linecorp/armeria/common/auth/OAuth1aTokenBuilder.java

```diff
@@ -0,0 +1,148 @@
+
+/**
+ * Builds a new {@link OAuth1aToken}.
+ */
+public final class OAuth1aTokenBuilder {
+
+    private static final String DEFAULT_OAUTH_VERSION = "1.0";
+
+    @Nullable
+    private String consumerKey;
+    @Nullable
+    private String token;
+    @Nullable
+    private String signatureMethod;
+    @Nullable
+    private String signature;
+    @Nullable
+    private String timestamp;
+    @Nullable
+    private String nonce;
+    @Nullable
+    private String realm;
+    private String version = DEFAULT_OAUTH_VERSION;
+    private Map<String, String> additionals = ImmutableMap.of();
+
+    /**
+     * Creates a new instance.
+     */
+    OAuth1aTokenBuilder() {}
+
+    /**
+     * Sets the value of the realm property.
+     */
+    public OAuth1aTokenBuilder realm(String realm) {
+        this.realm = requireNonNull(realm, "realm");
+        return this;
+    }
+
+    /**
+     * Sets the value of the oath_consumer_key property.
+     */
+    public OAuth1aTokenBuilder consumerKey(String consumerKey) {
+        this.consumerKey = requireNonNull(consumerKey, "consumerKey");
+        return this;
+    }
+
+    /**
+     * Sets the value of the oauth_token property.
+     */
+    public OAuth1aTokenBuilder token(String token) {
+        this.token = requireNonNull(token, "token");
+        return this;
+    }
+
+    /**
+     * Sets the value of oauth_signature_method property.
+     */
+    public OAuth1aTokenBuilder signatureMethod(String signatureMethod) {
+        this.signatureMethod = requireNonNull(signatureMethod, "signatureMethod");
+        return this;
+    }
+
+    /**
+     * Sets the value of oauth_signature property.
+     */
+    public OAuth1aTokenBuilder signature(String signature) {
+        this.signature = requireNonNull(signature, "signature");
+        return this;
+    }
+
+    /**
+     * Sets the value of oauth_timestamp property.
+     */
+    public OAuth1aTokenBuilder timestamp(String timestamp) {
+        try {
+            Long.parseLong(requireNonNull(timestamp, "timestamp"));
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("timestamp: " + timestamp +
+                                               " (expected: a string containing a long representation)");
+        }
+        this.timestamp = timestamp;
+        return this;
+    }
+
+    /**
+     * Sets the value of oauth_nonce property.
+     */
+    public OAuth1aTokenBuilder nonce(String nonce) {
+        this.nonce = requireNonNull(nonce, "nonce");
+        return this;
+    }
+
+    /**
+     * Sets the value of oauth_version property.
+     * If not set, {@value DEFAULT_OAUTH_VERSION} is used by default.
+     */
+    public OAuth1aTokenBuilder version(String version) {
+        this.version = requireNonNull(version, "version");
+        return this;
+    }
+
+    /**
+     * Sets additional (or user-defined) parameters.
+     */
+    public OAuth1aTokenBuilder additionals(Map<String, String> additionals) {
```

Comment:
What happens if:

- a user puts existing properties such as `nonce` using this method?
- a user calls this method twice? Shouldn't we merge them?

When a user calls this method, we're gonna have to check if there's an entry with a known key, and if so, we should delegate to the specific setter method.

Reply:
That's a good suggestion. 👍